### PR TITLE
Config

### PR DESCRIPTION
**Pull request by lemmyg**
added the ability to define buttons layouts from the config.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -642,13 +628,13 @@ fn main() {
 
     PrivDrop::default()
         .user("nobody")
-        .group("nobody")
+        .group("nogroup")
```
Comment (lemmyg) [inline src/main.rs]: maybe I need to revert this change for ubuntu

File: Cargo.toml [hunk 2]
```diff
@@ -10,8 +10,8 @@ repository = "https://github.com/WhatAmISupposedToPutHere/tiny-dfr"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-cairo-rs = { version = "0.18", default-features = false }
-librsvg = { git = "https://gitlab.gnome.org/WhyHelloThere/librsvg.git", rev = "cb3d568d30207e0512f486a30001effabb80bb04", version = "2.57.0" }
+cairo-rs = { version = "0.17", default-features = false }
```
Comment (lemmyg) [inline Cargo.toml]: should I revert this change too?

File: src/main.rs [hunk 3]
```diff
@@ -484,17 +511,10 @@ struct UiConfig {
     primary_layer: LayerType,
     secondary_layer: LayerType,
     font: String,
-    icon_theme: String,
+    media_theme: String,
```
Comment (AdityaGarg8) [inline src/main.rs]: Shouldn’t here be app_icon_theme. Did you compile the daemon before making the change?

File: etc/tiny-dfr.conf [hunk 4]
```diff
@@ -1,15 +1,99 @@
...
-icon_theme = "tiny-dfr-icons"
-
-[apps]
-app_icon_theme = "tiny-dfr-icons"
-app1_icon = "app1"
-app2_icon = "app2"
-app3_icon = "app3"
-app4_icon = "app4"
+media_icon_theme = "tiny-dfr-icons"
+app_icon_theme = "breeze"
```
Comment (AdityaGarg8) [inline etc/tiny-dfr.conf]: App_icon_theme in the repo should be hicolor imo, it’s a theme available in all distros

**Review by AdityaGarg8 — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 3 above)
Comment (lemmyg) [reply src/main.rs]: good catch. I didnt test it as I need to revert some stuff.

File: etc/tiny-dfr.conf (hunk 4 above)
Comment (lemmyg) [reply etc/tiny-dfr.conf]: done

Comment (lemmyg) [reply etc/tiny-dfr.conf]: tested the code. working

File: etc/systemd/system/tiny-dfr.service [hunk 5]
```diff
@@ -7,3 +7,6 @@ StartLimitBurst=2
 [Service]
 ExecStart=/usr/bin/tiny-dfr
 Restart=always
+
+[Install]
```
Comment (lemmyg) [inline etc/systemd/system/tiny-dfr.service]: this was copied from ubuntu package.
Shall I remove it?

File: src/main.rs (hunk 1 above)
Comment (AdityaGarg8) [reply src/main.rs]: This fork is not only for Ubuntu, so not needed

File: Cargo.toml (hunk 2 above)
Comment (AdityaGarg8) [reply Cargo.toml]: No

File: etc/systemd/system/tiny-dfr.service (hunk 5 above)
Comment (AdityaGarg8) [reply etc/systemd/system/tiny-dfr.service]: Yes

File: src/main.rs [hunk 6]
```diff
@@ -107,8 +109,8 @@ impl Button {
                         image = ButtonImage::Png(image::open(icon_path_png).unwrap());
                 } else {
                     // If the icon is not found in /usr/share/pixmaps, use the icon_name as text
-                        let icon_name_label: &'static str = Box::leak(format!("{}", icon_name).into_boxed_str());
-                        image = ButtonImage::Text(icon_name_label);
+                        let icon_name_label = &Box::leak(format!("{}", icon_name).into_boxed_str());
```
Comment (AdityaGarg8) [inline src/main.rs:112]: Since text now used string, I guess `buttonimage::Text(icon_name.to_string());` can also work now.

**Comment by AdityaGarg8** [comment]
I've tested your code and it works good.

**Review by AdityaGarg8 — APPROVED** [review]

**Comment by lemmyg** [comment]
Nice!
